packaging/libaec-config.cmake

```diff
@@ -0,0 +1,105 @@
+#   SZIP_VERSION   - The version of the SZIP library which was found.
+#   SZIP_LIBRARIES - All the required libraries to make use of SZIP.
+#
+# and the following imported targets:
+#
+#   libaec::aec-shared  - The shared AEC library target (if it was built).
+#   libaec::sz-shared   - The shared SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec-static  - The static AEC library target (if it was built).
+#   libaec::sz-static   - The static SZIP compatible version of the AEC library
+#                         (if it was built).
+#   libaec::aec         - The (shared or static) AEC library target (according
+#                         to the value of libaec_USE_STATIC_LIBS).
+#   libaec::sz          - The (shared or static) SZIP compatible version of the
+#                         AEC library (according to the value of
+#                         libaec_USE_STATIC_LIBS).
+
+include(${CMAKE_CURRENT_LIST_DIR}/libaec-targets.cmake)
+
+# Alias static or shared targets depending on libaec_USE_STATIC_LIBS
+if (libaec_USE_STATIC_LIBS)
+  if (TARGET libaec::aec-static AND TARGET libaec::sz-static)
+    add_library(libaec::aec ALIAS libaec::aec-static)
+    add_library(libaec::sz ALIAS libaec::sz-static)
+  else ()
+    set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE
+      "STATIC versions of libaec::aec and libaec::sz not found.")
+    set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
+  endif ()
+else ()
+  if (TARGET libaec::aec-shared AND TARGET libaec::sz-shared)
+    add_library(libaec::aec ALIAS libaec::aec-shared)
+    add_library(libaec::sz ALIAS libaec::sz-shared)
+  else ()
+    set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE
+      "SHARED versions of libaec::aec and libaec::sz not found.")
+    set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
+  endif ()
+endif ()
+
+if (TARGET libaec::sz)
+  get_target_property(SZIP_INCLUDE_DIR
+    libaec::sz INTERFACE_INCLUDE_DIRECTORIES)
+  if (SZIP_INCLUDE_DIR)
+    # This might be a list. The first item is the directory with the
+    # "main" header.
+    list(GET SZIP_INCLUDE_DIR 0 SZIP_INCLUDE_DIR)
+  endif ()
+
+  # Loop over all configurations of libaec::sz and set SZIP_LIBRARIES
+  # to the first configuration with an existing file for
+  # IMPORTED_IMPLIB_<CONFIG> or, if that does not exist, a file for
+  # IMPORTED_LOCATION_<CONFIG>.
+  get_property(_isMultiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+  if (_isMultiConfig)
+    # For multi-configuration generators (e.g., Visual Studio), prefer
+    # those configurations.
+    string(TOUPPER "${CMAKE_CONFIGURATION_TYPES}" _build_types)
+  else ()
+    # For single-configuration generators, prefer the current
+    # configuration.
+    string(TOUPPER "${CMAKE_BUILD_TYPE}" _build_types)
+  endif ()
+  get_target_property(_imported_configs libaec::sz IMPORTED_CONFIGURATIONS)
+
+  foreach (_config ${_build_types})
+    if (_config IN_LIST _imported_configs)
```

Comment:
If I understand correctly, this condition could lead to cases where the variables aren't set even if the CMake targets are successfully imported.
E.g., if `libaec` was built (only) with the configuration "Release" and a project that imports this is being built with the configuration "Debug".
In that case, it would be ok to link to the "Release" binaries imho.

It would probably be better to just concatenate the `_build_types` list with the `_imported_configs` list (with `_build_types` being first) and iterate over that list.
Optionally, remove duplicates in that concatenated list with [`list(REMOVE_DUPLICATES _all_types)`](https://cmake.org/cmake/help/latest/command/list.html#remove-duplicates).

Reply:
Indeed, that would be better.